fontforge/featurefile.c

```diff
@@ -2435,21 +2438,43 @@ return;
 return;
     }
 
-    if ( *namebuf=='/' ||
-	    ( pt = strrchr(tok->filename[tok->inc_depth],'/') )==NULL )
-	filename=copy(namebuf);
-    else {
-	*pt = '\0';
-	filename = GFileAppendFile(tok->filename[tok->inc_depth],namebuf,false);
-	*pt = '/';
+
+    pt = strrchr(tok->filename[tok->inc_depth],'/');
+    if ( pt != NULL ) *pt = '\0';
+    filename_deprecated = GFileAppendFile(tok->filename[tok->inc_depth],namebuf,false);
+    if ( pt != NULL ) *pt = '/';
+    if ( tok->in_ufo ) {
+        // we know this is a features.fea file
+        filename = copy(tok->filename[tok->inc_depth]);
+        char* filename_above = GFileAppendFile(GFileDirName(filename), "..", false);
```

Comment:
This leaks the result from GFileDirName

Reply:
I think both these problems would be solved by just doing `filename = GFileDirName(tok->filename[tok->inc_depth]);` on the first line and `GFileAppendFile(filename, "..", false);` on the second.

Reply:
Whoops -- alter ego there.